Libplanet/Tx/ITxExcerpt.cs

```diff
@@ -0,0 +1,16 @@
+namespace Libplanet.Tx
+{
+    /// <summary>
+    /// Similar to <see cref="ITxMetadata"/> except that it has <see cref="TxId"/> as well.
+    /// Note that this does not contain actions or signature.
+    /// </summary>
+    public interface ITxExcerpt : ITxMetadata
```

Comment:
Can we clarify it's use a bit more?

As I understand it, seems like this is to "group" `UntypedTransaction` and `Transaction<T>` for validating nonces. In any case, the relation between `ITxExcerpt` and `ITxMetadata` should mirror that of `IBlockExcerpt` and `IBlockMetadata` if possible. 😕

Reply:
I defined the term *excerpt* as a set of metadata functioning as a proof without complete data (usually lacking its body like actions and transactions).  That's why `IBlockExcerpt` has `Hash`, and `ITxExcerpt` has `Id`.

Reply:
> I defined the term excerpt as a set of metadata

I see *excerpts* are some sort of metadata. On the other hand, `ITxMetadata` and `IBlockMetadata` have analogous specific uses. I'm still confused as to what this is "proving" really and if `ITxExcerpt : ITxMetadata` is necessary.

- If the term *excerpt* is to *extend* metadata interfaces with additional proof data, than inheritance is granted.
- If the term *excerpt* is to be used for naming an auxiliary `interface` with minimal amount of information than there shouldn't be an inheritance relation.

I'm suggesting that either we have `IBlockExcerpt : IBlockMetadata` and `ITxExcerpt : ITxMetadata` (this is the latter of the above) or not have inheritance relations between them (the former of the above). The split design seems confusing.

Reply:
Ah, I see.

Reply:
Not to pester, but is this being worked on? I'm done with ~skimming~ reviewing the code, but I wasn't sure about your last comment. 😶

Reply:
I tried to make `ITxMetadata` to inherit nothing, but it wasn't that easy as `TxExcerptExtensions.ValidateTxNonces()` used several properties of `ITxMetadata`.  Instead, I'm going to rename its name later.

Reply:
👌

Reply:
<https://github.com/planetarium/libplanet/issues/1997>